pkg/api/v1alpha1/tinkerbelltemplateconfig_defaults.go

```diff
@@ -21,76 +21,80 @@ warnings:
 `
 )
 
-func getDiskPart(disk string) string {
-	switch {
-	case strings.Contains(disk, "nvme"):
-		return fmt.Sprintf("%sp", disk)
-	default:
-		return disk
+// GetDefaultActionsFromBundle constructs a set of default actions for the given osFamily using the
+// bundle as the source of action images.
+func GetDefaultActionsFromBundle(b v1alpha1.VersionsBundle, osImageOverride, tinkerbellLocalIP, tinkerbellLBIP string, osFamily OSFamily) []ActionOpt {
+	// The metadata string will have two URLs:
+	// 1. one that will be used initially for bootstrap and will point to hegel running on kind.
+	// 2. one that will be used when the workload cluster is up and will point to hegel running on
+	//    the workload cluster.
+	metadataURLs := []string{
+		fmt.Sprintf("http://%s:50061", tinkerbellLocalIP),
+		fmt.Sprintf("http://%s:50061", tinkerbellLBIP),
 	}
-}
 
-func GetDefaultActionsFromBundle(b v1alpha1.VersionsBundle, disk, osImageOverride, tinkerbellLocalIp, tinkerbellLBIp string, osFamily OSFamily) []ActionOpt {
-	var diskPart string
+	// During workflow reconciliation when the Tinkerbell template is rendered, the Workflow
+	// Controller injects a subset of data from the Hardware resource. This lets us use Go template
+	// language to render the disks enabling mix'n'match disk types for templates that represent
+	// the same kind of machine such as control plane nodes.
+	//
+	// The devicePath disk index and the storagePartitionPath disk index should match.
+	const devicePath = "{{ index .Hardware.Disks 0 }}"
+	const paritionPathFmt = "{{ formatPartition ( index .Hardware.Disks 0 ) %s }}"
```

Comment:
So these are template strings now instead of the actual values so yaml can replace them? Also any reason why they are consts defined within a function vs just a variable, you want to avoid mutability of it? If so, why not define them above where it can be used by the unit tests as well?

Reply:
We can expand the variable names further, although I'm not sure I see the value in the addition given the scope of the variable.

I'm not quite following what the 'Condition' part is referencing, could you elaborate?

They're denoted as constants so they're allocated once. I don't think they need to be made part of the package API though, particularly given this is the API package and already bloated.

Reply:
For clarity. 'devicePath' is being used because it's used in place of various Tinkerbell template device path parameters. The fact it's a templated value felt of less importance.